scripts/gff_metaparser/gff_stats.py

```diff
@@ -28,89 +28,136 @@
-                      type=argparse.FileType('rt', encoding='UTF-8'),
-                      help="valid structures config file")
-  # various options
-  parser.add_argument("--stats_only", action="store_true", required=False, help="produce only stats output")
-  parser.add_argument("--no_id_stats", action="store_true", required=False, help="do not dump stemmed id stats")
-  parser.add_argument("--fail_unknown", action="store_true", required=False, help="fail if unknown structure met")
-  parser.add_argument("--processed_qualifiers",
-                      metavar = "source,name,parent,dbxref,phase,product,protein_id,biotype",
-                      required=False, type = str, default = "",
-                      help="comma separated list of qualifiers to use [keeping all by default]")
-  # ? separate sources, value to include #ALL_SOURCES with #
-  parser.add_argument("--no_longest_pfx", action="store_true", required=False, help="no lognest pfrefixes for IDs" )
-  parser.add_argument("--dump_used_options", action="store_true", required=False, help="dump used (not None) options")
-  parser.add_argument("--rule_options", metavar="load_pseudogene_with_CDS,option2,...",
-                     required=False, action="append",
-                     help="options to control conditional rules")
-  parser.add_argument("--no_contig_len_extenstion", action="store_true", required=False, help="do not extend contig length based on the feature boundaries (try to omit, if fails)")
-  # output
-  parser.add_argument("--stats_out", metavar="stats.out", required = False,
-                      type=argparse.FileType('w',  encoding='UTF-8'), default=sys.stdout,
-                      help="stats output [STDOUT]" )
-  parser.add_argument("--detailed_report", metavar="detailed.log", required = False,
-                      type=argparse.FileType('w',  encoding='UTF-8'),
-                      help="file to report every failed and modified model" )
-  parser.add_argument("--gff_out", metavar="out.gff3", required = False,
-                      type=argparse.FileType('w', encoding='UTF-8'), default=sys.stdout,
-                      help="resulting gff output [STDOUT]" )
-  # input
-  parser.add_argument("--fasta", metavar="fasta.fna", required = False,
-                      type=str,
-                      help="fasta file with sequences to calculate length (region length or max feature end will be used, if absent)")
-  parser.add_argument("gff_in", metavar="in.gff3",
-                      type=argparse.FileType('rt', encoding='UTF-8'),
-                      help="input gff file (use '-' to read from STDIN)" )
-  args = parser.parse_args()
-  if args.dump_used_options:
-    print("used options:", file=sys.stderr)
-    for o, v in args.__dict__.items():
-      if v: print("  %s: %s" %(o, v), file=sys.stderr)
-  return args
+    parser = argparse.ArgumentParser()
+    # configs
+    parser.add_argument(
```

Comment:
argparse args style

Reply:
no need to comment every instance of this reformatting, we can discuss it globally